### PR TITLE
Update build missing warning message

### PR DESCRIPTION
**Pull request by uilianries**
Update missing package warning message:
 - Using CCI package czmq for demonstration;
 - Added the new information: Requirements;
 - There is only one occurrence of that warning message in docs;
 - As semver_direct_mode is not affected by minor and patch version, all requirements show placeholders.
 
Related issue: https://github.com/conan-io/conan/pull/6700

### CODE REVIEW

File: faq/troubleshooting.rst [hunk 1]
```diff
@@ -10,25 +10,28 @@ that you get an error like the following one:
...
+    - Options: shared=False, with_libcurl=True, with_libuuid=True, with_lz4=True, libcurl:shared=False, libcurl:with_brotli=False, libcurl:with_largemaxwritesize=False, libcurl:with_ldap=False, libcurl:with_libidn=False, libcurl:with_libmetalink=False, libcurl:with_libpsl=False, libcurl:with_librtmp=False, libcurl:with_libssh2=False, libcurl:with_nghttp2=False, libcurl:with_openssl=True, libcurl:with_winssl=False, libsodium:PIE=False, libsodium:shared=False, libsodium:use_soname=True, lz4:shared=False, openssl:386=False, openssl:capieng_dialog=False, openssl:no_asm=False, openssl:no_async=False, openssl:no_bf=False, openssl:no_cast=False, openssl:no_des=False, openssl:no_dh=False, openssl:no_dsa=False, openssl:no_dso=False, openssl:no_hmac=False, openssl:no_md2=False, openssl:no_md5=False, openssl:no_mdc2=False, openssl:no_rc2=False, openssl:no_rc4=False, openssl:no_rc5=False, openssl:no_rsa=False, openssl:no_sha=False, openssl:no_sse2=False, openssl:no_threads=False, openssl:no_zlib=False, openssl:openssldir=None, openssl:shared=False, zeromq:encryption=libsodium, zeromq:shared=False, zlib:minizip=False, zlib:shared=False
```
Comment (uilianries) [inline faq/troubleshooting.rst]: Maybe we could drop some options for this demonstration, is visual polluted IMO.

Comment (danimtb) [reply faq/troubleshooting.rst]: yep, I think it is too much. You can crop it and add some `...` for new lines. WDYT?

Comment (memsharded) [reply faq/troubleshooting.rst]: Yes, please crop and use some ``...`` or ``... <more options>...``

Comment (uilianries) [reply faq/troubleshooting.rst]: done, much clear now.

File: faq/troubleshooting.rst [hunk 2]
```diff
@@ -10,25 +10,28 @@ that you get an error like the following one:
...
+    Or read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package"
 
 
-This means that the package recipe ``libzmq/4.2.0@memsharded/testing`` exists, but for some reason
+This means that the package recipe ``czmq/4.2.0@`` exists, but for some reason
 there is no precompiled package for your current settings. Maybe the package creator didn't build
 and shared pre-built packages at all and only uploaded the package recipe, or maybe they are only
 providing packages for some platforms or compilers. E.g. the package creator built packages
-from the recipe for gcc 4.8 and 4.9, but you are using gcc 5.4.
+from the recipe for Visual Studio 14 and 15, but you are using gcc 16.
```
Comment (jgsogo) [inline faq/troubleshooting.rst]:
```suggestion
from the recipe for Visual Studio 14 and 15, but you are using Visual Studio 16.
```

**Comment by memsharded** [comment]
@jgsogo PRs keep failing for unrelated links checks. I suggested completely removing them, leaving them only in nightlies or similar.

**Comment by jgsogo** [comment]
Those links are actually broken

**Comment by memsharded** [comment]
> Those links are actually broken

Still, not the fault of this PR, and this kind of errors happen on PR a bit too much for my taste, some unrelated link to the PR that doesn't respond and breaks the PR contribution, which cannot longer be merged even if it is clear that the error is unrelated.

**Comment by jgsogo** [comment]
My main concern is not realizing that these links are broken and failing in `master` (now it is broken, did anyone realize that? what is missing in the docs and for how long?). We will merge a PR, think that everything is done and the docs are published, but no, because of the failure we don't update the docs. I prefer it to fail in the PR than to fail in `master`.

This check has always been there and from time to time there are broken links we need to fix. It is not something new. I agree it is PITA, I would vote to remove this check, but we need something to ensure that `master` is green/published. Something that forces us to fix broken links from time to time.

**Pull request merged**